Tests fail to build on ghc 9.2.1

Hey, getting the following:

```
test/TestBarbies.hs:256:19: error:
    • Non type-variable argument in the constraint: Show (Record3 f)
      (Use FlexibleContexts to permit this)
    • In the stand-alone deriving instance for
        ‘(Show (f Int), Show (Record3 f)) => Show (NestedF f)’
    |
256 | deriving instance (Show (f Int), Show (Record3 f)) => Show (NestedF f)
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

test/TestBarbies.hs:257:19: error:
    • Non type-variable argument in the constraint: Eq (Record3 f)
      (Use FlexibleContexts to permit this)
    • In the stand-alone deriving instance for
        ‘(Eq (f Int), Eq (Record3 f)) => Eq (NestedF f)’
    |
257 | deriving instance (Eq   (f Int), Eq   (Record3 f)) => Eq   (NestedF f)
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[10 of 12] Compiling TestBarbiesW     ( test/TestBarbiesW.hs, dist/build/barbies-test/barbies-test-tmp/TestBarbiesW.o )

test/TestBarbiesW.hs:271:19: error:
    • Non type-variable argument
        in the constraint: Show (Record3W Covered f)
      (Use FlexibleContexts to permit this)
    • In the stand-alone deriving instance for
        ‘(Show (f Int), Show (Record3W Covered f)) =>
         Show (NestedFW Covered f)’
    |
271 | deriving instance (Show (f Int), Show (Record3W Covered f)) => Show (NestedFW Covered f)
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

test/TestBarbiesW.hs:272:19: error:
    • Non type-variable argument
        in the constraint: Eq (Record3W Covered f)
      (Use FlexibleContexts to permit this)
    • In the stand-alone deriving instance for
        ‘(Eq (f Int), Eq (Record3W Covered f)) => Eq (NestedFW Covered f)’
    |
272 | deriving instance (Eq   (f Int), Eq   (Record3W Covered f)) => Eq   (NestedFW Covered f)
    |
```

Should just be a matter of turning FlexibleContexts on - it's a bug that got fixed with ghc afaik.

--- a/test-legacy/Legacy/TestBarbies.hs
+++ b/test-legacy/Legacy/TestBarbies.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE FlexibleContexts     #-}
 {-# LANGUAGE TypeFamilies         #-}
 {-# LANGUAGE UndecidableInstances #-}
 module Legacy.TestBarbies

--- a/test-legacy/Legacy/TestBarbiesW.hs
+++ b/test-legacy/Legacy/TestBarbiesW.hs
@@ -1,5 +1,6 @@
 {-# OPTIONS_GHC -O0 #-}
 {-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE FlexibleContexts     #-}
 {-# LANGUAGE FlexibleInstances    #-}
 {-# LANGUAGE TypeFamilies         #-}
 {-# LANGUAGE UndecidableInstances #-}

--- a/test/TestBarbies.hs
+++ b/test/TestBarbies.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE FlexibleContexts     #-}
 {-# LANGUAGE TypeFamilies         #-}
 {-# LANGUAGE UndecidableInstances #-}
 module TestBarbies

--- a/test/TestBarbiesW.hs
+++ b/test/TestBarbiesW.hs
@@ -1,5 +1,6 @@
 {-# OPTIONS_GHC -O0 #-}
 {-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE FlexibleContexts     #-}
 {-# LANGUAGE FlexibleInstances    #-}
 {-# LANGUAGE TypeFamilies         #-}
 {-# LANGUAGE UndecidableInstances #-}

--- a/test/TestBiBarbies.hs
+++ b/test/TestBiBarbies.hs
@@ -1,5 +1,6 @@
 {-# LANGUAGE AllowAmbiguousTypes  #-}
 {-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE FlexibleContexts     #-}
 {-# LANGUAGE PolyKinds            #-}
 {-# LANGUAGE TypeFamilies         #-}
 {-# LANGUAGE UndecidableInstances #-}

